dev/run-tests.py

```diff
@@ -323,7 +323,7 @@ def get_hadoop_profiles(hadoop_version):
 def build_spark_maven(hadoop_version):
     # Enable all of the profiles for the build:
     build_profiles = get_hadoop_profiles(hadoop_version) + modules.root.build_profile_flags
-    mvn_goals = ["clean", "package", "-DskipTests"]
+    mvn_goals = ["clean", "package", "-DskipTests", "-pl", "!assembly"]
```

Comment:
If you're looking at speeding up the build, building and testing in one shot might be a good thing to do at least for maven. Using `-fae` would allow the build to go as far as it can when something fails.

No need to do that in this change, though.